API/contact/ContactAPI.php

```diff
@@ -0,0 +1,148 @@
+         * Creates contact record.
+         * 
+         * @param object $contact object of the Contact class.
+         * @return object|false object of the Contact class containing all information about created record or false if operation was unsuccessful. 
+         */
+        public function CreateContact(object $contact) : object|false 
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+
+            $stmt = $this->mysql->prepare("INSERT INTO Contacts (ID, firstName, lastName, email, phone, userID) VALUES(DEFAULT, ?, ?, ?, ?, ?)");
+            $stmt->bind_param(
+                "sssss", 
+                $contact->firstName, 
+                $contact->lastName, 
+                $contact->email, 
+                $contact->phone, 
+                $contact->userID
+            );
+            
+            $result = $stmt->execute();
+
+            if ($result)
+                return $this->GetContactByID($this->mysql->insert_id);
+
+            return false;
+        }
+
+        /**
+         * Gets contact record by contact's unique identifier.
+         * 
+         * @param int $contactID unique contact identifier.
+         * @return object|false object of the Contact class containing all information about record or false if operation was unsuccessful.
+         */
+        private function GetContactByID($contactID) : object|false
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+
+            $result = $this->mysql->query("SELECT * FROM Contacts WHERE ID=$contactID");
+
+            if ($result == false)
+                return false;
+        
+            $record = $result->fetch_object();
+
+            if ($record == null)
+                return false;
+
+            return Contact::Deserialize($record);
+        }
+
+        /**
+         * Gets contact record which satisfies query.
+         * 
+         * @param string $query search query used for searching for set of records in the database.
+         * @param int $numOfResults max number of results that satisfy search query to be returned if search is successful.
+         * @return array|false array of objects of the Contact class containing all information about each individual record or false if operation was unsuccessful.
+         */
+        public function GetContact(string $query, int $numOfResults) : array|false
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+
+            $queryArray = explode(" ", $query);
+
+            $searchQuery = "";
+
+            foreach ($queryArray as $word) {
+                $searchQuery = $searchQuery . 
+                "firstName LIKE '%$word%' OR 
+                lastName LIKE '%$word%' OR
+                phone LIKE '%$word%' OR
+                email LIKE '%$word%'" . " OR ";
+            }
+
+            $searchQuery = substr($searchQuery, 0, strlen($searchQuery) - 4);
```

Comment:
What is the -4 doing here? Can you please add a comment to make it more clear?

Reply:
next commit will address that